lib/view_component/base.rb

```diff
@@ -66,7 +66,7 @@ def render_in(view_context, &block)
       # Assign captured content passed to component as a block to @content
       @content = view_context.capture(self, &block) if block_given?
 
-      before_render_check
+      before_render
```

Comment:
I'm wondering if `before_render` should happen before we assign `@content`?

Slightly related, I wonder if we should only call `view_context.capture(self, &block)` if `render?` is true?

Reply:
Good question. I can see benefits to it being called in both orders of execution.

When I was writing this PR, I was intending `before_render` to mean "once ViewComponent has done all the magic to get the component to render, but before it actually renders."

Could imagine someone using this hook to set an instance variable based on whether `@content` is present or not, for example.

Similarly, I do think that `render?` could execute before `before_render`, but I'd be hesitant to move it to before capturing the provided block, as in theory folks could be using `render?` to check for the presence of `@content`.

Reply:
Yeah, I was thinking a bit about it and I think that makes sense.

I think being able to check the presence of `@content` to ensure a block was passed would make sense, but I wonder if it could lead to people checking the contents of `@content` which I could see being either (or both) a feature and an anti-pattern.